### PR TITLE
CO-3693 add warning message when no child were found with given criteria

### PR DESCRIPTION
**Pull request by JonathanGuerne**
- raise an error in the controller if no child that could match the criteria where found
- if and error was raise by the controller it displays a warning message to the user for 5seconds

- new sponsor button is removed
  - it only clear name, last name and email field. it doesn't seem very useful
  -  the name of the button is confusing
    - the button could be kept but with an other text (like "clear")

### CODE REVIEW

File: sms_sponsorship/webapp/src/i18n.js [hunk 1]
```diff
@@ -35,7 +36,6 @@ i18n
                     "emailConfirm": "Confirm your email address",
                     "sponsorshipPlus": "Sponsorship plus",
                     "otherChild": "Other child",
-                    "newSponsor": "New sponsor",
```
Comment (ecino) [inline sms_sponsorship/webapp/src/i18n.js:38]: Why is this string removed?

File: sms_sponsorship/webapp/src/i18n.js [hunk 2]
```diff
@@ -94,6 +94,7 @@ i18n
                     "waitingForChild": "Merci de patienter quelques secondes pendant que nous recherchons un enfant en attente d'un parrain...",
                     "waitingForOtherChild": "Nous recherchons un enfant qui correspond à vos critères. Merci de patienter quelques secondes...",
                     "ageYears": " ans",
+                    "notMatchCriteria": "Could not found a child with given criteria.",
```
Comment (ecino) [inline sms_sponsorship/webapp/src/i18n.js]: This should be in French.

File: sms_sponsorship/webapp/src/i18n.js [hunk 3]
```diff
@@ -164,6 +164,7 @@ i18n
                     "waitingForChild": "Bitte warte ein paar Sekunden, wir suchen ein Kind, das auf einen Paten wartet ...",
                     "waitingForOtherChild": "Wir suchen ein Kind, das zu dir passt, bitte warte ein paar Sekunden ...",
                     "ageYears": " Jahre",
+                    "notMatchCriteria": "Could not found a child with given criteria.",
```
Comment (ecino) [inline sms_sponsorship/webapp/src/i18n.js]: Could you enter the translations please? Deepl should work fine for that (maybe check German with Andre).

File: sms_sponsorship/webapp/src/i18n.js [hunk 4]
```diff
@@ -234,6 +234,7 @@ i18n
                     "waitingForChild": "Stiamo cercando un bambino in attesa di un sostenitore, per favore aspetta qualche secondo ...",
                     "waitingForOtherChild": "Stiamo cercando un bambino che corrisponda alla tua richiesta, per favore aspetta qualche secondo ...",
                     "ageYears": " anni",
+                    "notMatchCriteria": "Could not found a child with given criteria.",
```
Comment (ecino) [inline sms_sponsorship/webapp/src/i18n.js]: Use deepl to put it in Italian.

**Review by ecino — CHANGES_REQUESTED** [review]

File: sms_sponsorship/webapp/src/i18n.js (hunk 1 above)
Comment (JonathanGuerne) [reply sms_sponsorship/webapp/src/i18n.js:38]: This text was related to the "new sponsor" button that I propose to remove
it only reset the content of the form. If you found it mandatory I would suggest to change the button's text as it is a bit confusing imo

Comment (ecino) [reply sms_sponsorship/webapp/src/i18n.js:38]: OK

**Review by ecino — APPROVED** [review]

**Pull request merged**